src/libasr/runtime/lfortran_intrinsics.c

```diff
@@ -1202,3 +1255,325 @@ LFORTRAN_API int32_t _lpython_get_argc() {
+    char *file_contents = _lpython_read(fd, size);
+    _lpython_close(fd);
+    free(filename);
+
+    char s[LCOMPILERS_MAX_STACKTRACE_LENGTH];
+    bool address = true;
+    uint32_t j = 0;
+    d->stack_size = 0;
+    for (uint32_t i = 0; i < size; i++) {
+        if (file_contents[i] == '\n') {
+            memset(s, '\0', sizeof(s));
+            j = 0;
+            d->stack_size++;
+            continue;
+        } else if (file_contents[i] == ' ') {
+            s[j] = '\0';
+            j = 0;
+            if (address) {
+                d->addresses[d->stack_size] = strtol(s, NULL, 10);
+                address = false;
+            } else {
+                d->line_numbers[d->stack_size] = strtol(s, NULL, 10);
+                address = true;
+            }
+            memset(s, '\0', sizeof(s));
+            continue;
+        }
+        s[j++] = file_contents[i];
+    }
+}
+
+char *read_line_from_file(char *filename, uint32_t line_number) {
+    FILE *fp;
+    char *line = NULL;
+    size_t len = 0, n = 0;
+
+    fp = fopen(filename, "r");
+    if (fp == NULL) exit(1);
+    while (n < line_number && (getline(&line, &len, fp) != -1)) n++;
+    fclose(fp);
+
+    return line;
+}
+
+static inline uint64_t bisection(const uint64_t vec[],
+        uint64_t size, uint64_t i) {
+    if (i < vec[0]) return 0;
+    if (i >= vec[size-1]) return size;
+    uint64_t i1 = 0, i2 = size-1;
+    while (i1 < i2-1) {
+        uint64_t imid = (i1+i2)/2;
+        if (i < vec[imid]) {
+            i2 = imid;
+        } else {
+            i1 = imid;
+        }
+    }
+    return i1;
+}
+
+char *remove_whitespace(char *str) {
+    char *end;
+    // remove leading space
+    while(isspace((unsigned char)*str)) str++;
+    if(*str == 0) // All spaces?
+        return str;
+    // remove trailing space
+    end = str + strlen(str) - 1;
+    while(end > str && isspace((unsigned char)*end)) end--;
+    // Write new null terminator character
+    end[1] = '\0';
+    return str;
+}
+
+int generate_stacktrace_files() {
+    // TODO: Replace the hardcoded part
+    char *base_name = get_base_name(source_filename);
+    char *cmd = malloc(strlen(base_name)*7 + 200);
+#ifdef HAVE_LFORTRAN_MACHO
+    strcpy(cmd, "dsymutil ");
+    strcat(cmd, base_name);
+    strcat(cmd, ".out && llvm-dwarfdump --debug-line ");
+    strcat(cmd, base_name);
+    strcat(cmd, ".out.dSYM > ");
+#else
+    strcpy(cmd, "llvm-dwarfdump --debug-line ");
+    strcat(cmd, base_name);
+    strcat(cmd, ".out > ");
+#endif
+    strcat(cmd, base_name);
+    strcat(cmd, "_ldd.txt && (cd src/bin; ./dwarf_convert.py ../../");
+    strcat(cmd, base_name);
+    strcat(cmd, "_ldd.txt ../../");
+    strcat(cmd, base_name);
+    strcat(cmd, "_lines.txt ../../");
+    strcat(cmd, base_name);
+    strcat(cmd, "_lines.dat && ./dat_convert.py ../../");
+    strcat(cmd, base_name);
+    strcat(cmd, "_lines.dat)");
+    int status = system(cmd);
+    free(cmd);
+    return status;
+}
```

Comment:
Let's do this at compile time. Let's do it by default when `-g` is present. Later we can provide another flag to turn this generation off, if needed.